### DISCUSSION TITLE
Why does kitty resize smoothly under Gnome X11 and jitter during resize everywhere else?

### DISCUSSION THREAD

**Discussion by neurocyte** [kovidgoyal/kitty#6928] [General]
Resizing the kitty window on Gnome X11 gives really, really smooth results. The output is perfectly synchronized with the redraw inside the terminal and no jumping or word wrapping is visible.

https://github.com/kovidgoyal/kitty/assets/1552770/47313192-3064-4d00-8c39-c1bec190b2f9

On every other DM I tested there are significant resizing artifacts. The image jumps up and down and word wrapping may be visible. Here is a video of I3 (also X11). I've tested this on I3, Hyprland, River and even Gnome Wayland and they all show similar or identical results.

https://github.com/kovidgoyal/kitty/assets/1552770/aa743b50-2d42-476d-a257-a79f09109096

What is so special about Gnome X11? It's the same install of Kitty and the same TUI application. The "Current kitty options" page is identical (at least for variations running on X11).

I would love to find a fix for I3. Adding (or removing) a compositor does not seem to make any difference.

**Comment by kovidgoyal** [comment]
See resize_debounce_time in kitty.conf. I will note that with the default settings resize is smooth on my i3 based system when vim is the terninal program. Hard to tell exactly whats happening in your video but if you grep the code for resize_debounce you shoud be able to add some prints and see. Basically, for a smooth resize you dont want kitty to tell the terminal program to resize itself until after you are done resizing.
